### ISSUE TITLE
Switch to cibuildwheel for binary wheels?

### ISSUE THREAD

**Issue by icemac** [zopefoundation/meta#55]
I came across this blog post https://fruch.github.io/2020/12/29/github-actions describing that they are using https://github.com/joerick/cibuildwheel to build binary wheels. Maybe this could ease our setup, too.

**Comment by dataflake** [comment]
I just came across something else from Hanno that I had completely forgotten about: https://github.com/hannosch/zope_wheels

This uses a Docker container specifically made for building `manylinux` wheels (https://quay.io/repository/pypa/manylinux1_x86_64) to build wheels in bulk from the last tag on a list of repositories. It's Linux only, no Windows, though.

**Comment by dataflake** [comment]
@icemac This can be closed with the GHA-based build stretegy, right?

**Comment by icemac** [comment]
No, we could still use `cibuildwheel`, especially as it has support for e. E. macOS Apple Silicon (I saw in a PR in a repository (I do not remember which one) that adding the wheels for this M1 chip was just adding a parameter to an env variable).
I think it could make building the binary wheels easier but I never tried it.

**Comment by icemac** [comment]
`cibuildwheel` already has support for Python 3.10. We are using https://github.com/matthew-brett/multibuild to install missing Python versions on AppVeyor. As it uses https://www.python.org/ftp/python to download the needed Python version, it has no chance as there is no Python 3.10 version uploaded there, yet.

**Comment by henryiii** [comment]
FYI, you have to use the 2.0 beta (should be made into a final release in about a week https://github.com/pypa/cibuildwheel/issues/657) and you have to enable CIBW_PRERELEASE_PYTHONS, because the API is not declared stable yet (actually, 3.10b4, which just came out and is [about to land in cibuildwheel](https://github.com/pypa/cibuildwheel/issues/657), just waiting on manylinux), is I think [supposed to be](https://github.com/pypa/manylinux/issues/949#issuecomment-835831841) the first ABI stable release.

**Comment by icemac** [comment]
I have a prototype here: https://github.com/zopefoundation/ExtensionClass/pull/39 – at least it looks simpler than the currently existing version and it includes Windows wheels in GHA.

**Comment by icemac** [comment]
Currently we have a robust enough way to create wheels. So closing this issue for now.

**Comment by henryiii** [comment]
FYI, since this discussion three years ago, cibuildwheel has joined the PyPA and become the standard way to build wheels for most projects, handling lots of special cases, like the correct min version of macOS for 3.12 and 3.13.
